File: test/Zametek.Maths.Graphs.Primitives.Tests/Entities/WorkStreamTests.cs
using Shouldly;
using System;
using Xunit;

namespace Zametek.Maths.Graphs.Tests
{
    public class WorkStreamTests
    {
        [Fact]
        public void WorkStream_GivenCtor_ThenPropertiesSet()
        {
            var workStream = new WorkStream<int>(1, "phase1", isPhase: true);

            workStream.Id.ShouldBe(1);
            workStream.Name.ShouldBe("phase1");
            workStream.IsPhase.ShouldBeTrue();
        }

        [Fact]
        public void WorkStream_GivenCloneObject_ThenAllPropertiesPreserved()
        {
            var workStream = new WorkStream<int>(1, "phase1", isPhase: true);

            var clone = (WorkStream<int>)workStream.CloneObject();

            clone.Id.ShouldBe(1);
            clone.Name.ShouldBe("phase1");
            clone.IsPhase.ShouldBeTrue();
        }
    }
}

File: rust/primitives/tests/work_stream_tests.rs
//! Ports of `.../Entities/WorkStreamTests.cs`.

use zametek_maths_graphs_primitives::WorkStream;

#[test]
fn work_stream_given_ctor_then_properties_set() {
    let work_stream = WorkStream::<i32>::new(1, "phase1", true);
    assert_eq!(work_stream.id, 1);
    assert_eq!(work_stream.name, "phase1");
    assert!(work_stream.is_phase);
}

#[test]
fn work_stream_given_clone_then_all_properties_preserved() {
    let work_stream = WorkStream::<i32>::new(1, "phase1", true);
    let clone = work_stream.clone();
    assert_eq!(clone.id, 1);
    assert_eq!(clone.name, "phase1");
    assert!(clone.is_phase);
}
